### PR TITLE
Reinsurance csm

### PR DESCRIPTION
**Pull request by dcolleoni**

### CODE REVIEW

File: ifrs17-template/Import/InteractWithImportScopes/EvaluateImportScopes.ipynb
Comment (dcolleoni) [inline ifrs17-template/Import/InteractWithImportScopes/EvaluateImportScopes.ipynb:1]: reset changes

File: ifrs17-template/Report/Reports.ipynb [hunk 1]
```diff
@@ -58,7 +58,7 @@
         {
             "cell_type": "code",
             "source": [
-                "#!eval-notebook \"../Import/CloseImportTemplate\""
+                "#!import \"../Import/CloseImportTemplate\""
```
Comment (dcolleoni) [inline ifrs17-template/Report/Reports.ipynb:61]: reset

File: ifrs17-template/Report/Reports.ipynb [hunk 2]
```diff
@@ -92,16 +102,27 @@
             "execution_count": 0,
             "outputs": []
         },
+        {
+            "cell_type": "code",
```
Comment (dcolleoni) [inline ifrs17-template/Report/Reports.ipynb:106]: remove

File: ifrs17-template/Report/Reports.ipynb [hunk 3]
```diff
@@ -92,16 +102,27 @@
...
-                "\npv.ReportingNode = \"CH\";",
-                "\npv.ReportingPeriod = (2021, 3);",
+                "\npv.ReportingNode = reportingNode;",
+                "\npv.ReportingPeriod = reportingPeriod;",
                 "\npv.CurrencyType = CurrencyType.Contractual;",
                 "\npv.ColumnSlices = new string[]{};//\"GroupOfContract\", \"AmountType\", \"Scenario\", \"ServicePeriod\"",
-                "\npv.Scenario = null; //\"All\";",
-                "\npv.DataFilter = null; //new [] {(\"GroupOfContract\", \"DT1.2\"),(\"LiabilityType\", \"LIC\") };",
+                "\npv.Scenario = \"All\"; //\"Delta\";",
+                "\npv.DataFilter = new [] {(\"EconomicBasis\", \"L\") }; //new [] {(\"GroupOfContract\", \"DT1.2\"),(\"LiabilityType\", \"LIC\") };",
```
Comment (dcolleoni) [inline ifrs17-template/Report/Reports.ipynb:125]: reset filters

File: ifrs17/Import/Importers.ipynb [hunk 4]
```diff
@@ -1430,24 +1429,28 @@
...
+                "\n                accidentYear = (int?)parsedAccidentYear;",
+                "\n            }",
+                "\n",
                 "\n            (string AmountType, string EstimateType) valueType = datarow.ParseAmountAndEstimateType(ImportFormats.Cashflow, parsingStorage.DimensionsWithExternalId, parsingStorage.EstimateType, parsingStorage.AmountType);",
                 "\n            var values = datarow.Table.Columns.Where(c => c.ColumnName.StartsWith(nameof(RawVariable.Values))).OrderBy(c => c.ColumnName.Length).ThenBy(c => c.ColumnName)",
                 "\n                                .Select(x => datarow.Field<string>(x.ColumnName).CheckStringForExponentialAndConvertToDouble()).ToArray();",
-                "\n            ",
+                "\n",
+                "\n            // Extra adjustment to values",
+                "\n            values = values.AdjustValues(args, dataNodeData, accidentYear);",
```
Comment (dcolleoni) [inline ifrs17/Import/Importers.ipynb:1446]: this is not needed for Reinsurance csm

File: ifrs17/Utils/Extensions.ipynb [hunk 5]
```diff
@@ -177,8 +177,9 @@
                 "public static double CheckStringForExponentialAndConvertToDouble (this string s)",
                 "\n{   ",
                 "\n    if (s == null) return default;",
-                "\n    if (double.TryParse(s, NumberStyles.Number, CultureInfo.InvariantCulture, out var doubleValue)) return doubleValue;",
-                "\n    else { ApplicationMessage.Log(Error.ParsingInvalidOrScientificValue, s); return 1; }",
+                "\n    if (double.TryParse(s, NumberStyles.Number | NumberStyles.Float, CultureInfo.InvariantCulture, out var doubleValue)) return doubleValue;",
```
Comment (dcolleoni) [inline ifrs17/Utils/Extensions.ipynb:180]: This is to allow exponential format. Reset.

File: ifrs17/Import/4ImportScope-TechnicalMargin.ipynb [hunk 6]
```diff
@@ -581,9 +631,14 @@
         {
             "cell_type": "markdown",
             "source": [
-                "# Technical Margin allocation for Reinsurance",
+                "# Loss Recovery Component",
                 "\n",
-                "\nIn the case of Reinsurance a Loss Recovery Component is computed when the underlying gross business opens a Loss component. Loss Recovery Component uses the notion of Loss Recovery Component Boundary ([LoReCo Boundary](#loreco-boundary)). "
+                "\nIn the case of Reinsurance a Loss Recovery Component is computed when the underlying gross business opens a Loss component. ",
```
Comment (dtrzesniak) [inline ifrs17/Import/4ImportScope-TechnicalMargin.ipynb:636]: Add this: In the case of Reinsurance a Loss Recovery Component is computed when the underlying gross **_LRC_** business opens a Loss component. _If the underlying gross business is LIC, there is no Loss Recovery Component, as there is also no Loss Component in the gross side._

**Review by dtrzesniak — APPROVED** [review]

**Pull request merged**